externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/dialect/OracleSQLDialect.scala

```diff
@@ -0,0 +1,147 @@
+    val statement =
+      connection.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)
+    if (connection.getAutoCommit) {
+      statement.setFetchSize(fetchSize)
+    }
+    statement
+  }
+
+  override def getTablesQuery(
+      catalog: String,
+      schema: String,
+      tableName: String,
+      tableTypes: util.List[String]): String = {
+    val tTypes =
+      if (tableTypes == null || tableTypes.isEmpty) {
+        Set()
+      } else {
+        tableTypes.asScala.toSet
+      }
+    val query = new StringBuilder(
+      s"""SELECT OWNER  AS TABLE_SCHEMA,
+         |       TABLE_NAME,
+         |       TABLE_TYPE AS TABLE_TYPE
+         |FROM ALL_CATALOG
+         |""".stripMargin)
+
+    val filters = ArrayBuffer[String]()
+    if (StringUtils.isNotBlank(schema)) {
+      filters += s"OWNER LIKE '$schema'"
+    }
+
+    if (StringUtils.isNotBlank(tableName)) {
+      filters += s"$TABLE_NAME LIKE '$tableName'"
+    }
+
+    if (tTypes.nonEmpty) {
+      filters += s"(${
+          tTypes.map { tableType => s"$TABLE_TYPE = '$tableType'" }
+            .mkString(" OR ")
+        })"
+    }
+
+    if (filters.nonEmpty) {
+      query.append(" WHERE ")
+      query.append(filters.mkString(" AND "))
+    }
+
+    query.toString()
+  }
+
+  override def getTableTypesOperation(session: Session): Operation = {
+    throw KyuubiSQLException.featureNotSupported()
+  }
+
+  override def getColumnsQuery(
+      session: Session,
+      catalogName: String,
+      schemaName: String,
+      tableName: String,
+      columnName: String): String = {
+    val query = new StringBuilder(
+      """
+        |SELECT OWNER AS TABLE_SCHEMA
+        |     , TABLE_NAME
+        |     , COLUMN_NAME
+        |FROM ALL_TAB_COLUMNS
+        |""".stripMargin)
+
+    val filters = ArrayBuffer[String]()
+    if (StringUtils.isNotEmpty(schemaName)) {
+      filters += s"OWNER LIKE '$schemaName'"
+    }
+    if (StringUtils.isNotEmpty(tableName)) {
+      filters += s"$TABLE_NAME LIKE '$tableName'"
+    }
+    if (StringUtils.isNotEmpty(columnName)) {
+      filters += s"$COLUMN_NAME LIKE '$columnName'"
+    }
+
+    if (filters.nonEmpty) {
+      query.append(" WHERE ")
+      query.append(filters.mkString(" AND "))
+    }
+
+    query.toString()
+  }
+
+  override def getFunctionsOperation(session: Session): Operation = {
+    throw KyuubiSQLException.featureNotSupported()
+  }
+
+  override def getPrimaryKeysOperation(session: Session): Operation = {
+    throw KyuubiSQLException.featureNotSupported()
+  }
+
+  override def getCrossReferenceOperation(session: Session): Operation = {
+    throw KyuubiSQLException.featureNotSupported()
+  }
+
+  override def getTRowSetGenerator(): JdbcTRowSetGenerator = new OracleTRowSetGenerator
+
+  override def getSchemaHelper(): SchemaHelper = {
+//    throw KyuubiSQLException.featureNotSupported()
+    new OracleSchemaHelper
+  }
+
+  override def name(): String = {
+    "oracle"
+  }
```

Comment:
Scala allows to eliminate method's curly braces for single stmt body
```suggestion
  override def name(): String = "oracle"
```